framework/include/base/Factory.h

```diff
@@ -251,13 +251,7 @@ Factory::createUnique(const std::string & obj_name,
                       const InputParameters & parameters,
                       const THREAD_ID tid)
 {
-  auto object = createUnique(obj_name, name, parameters, tid, false);
-  if (!dynamic_cast<T *>(object.get()))
-    mooseError("We expected to create an object of type '" + demangle(typeid(T).name()) +
-               "'.\nInstead we received a parameters object for type '" + obj_name +
-               "'.\nDid you call the wrong \"add\" method in your Action?");
-
-  return std::unique_ptr<T>(static_cast<T *>(object.release()));
+  return std::unique_ptr<T>(create<T>(obj_name, name, parameters, tid).get());
```

Comment:
this doesn't release the shared ptr

Reply:
oh, I guess it does when it goes out of scope... ugh this is so dirty

Reply:
wait... no. this doesn't make sense. you'll get a ~shared_ptr when this leaves scope and the end up with a unique_ptr that points to... a destructed object

Reply:
uhm... yeah, dang